app/controllers/application_controller.rb

```diff
@@ -36,11 +36,12 @@ def after_sign_out_path_for(resource_or_scope)
   end
 
   def authenticate_with_basic_http
-    if %w(staging).include?(Rails.env) && request.path !~ /\A\/api.*/
-      unless session[:authenticated]
-        authenticate_or_request_with_http_basic 'Staging' do |name, password|
-          name == ENV["BASIC_HTTP_USER"] && password == ENV["BASIC_HTTP_PASSWORD"] && (session[:authenticated] = true)
-        end
+    return unless Rails.configuration.basic_auth_required
+    return if request.path =~ /\A\/api.*/
```

Comment:
We know this logic was already there in the application_controller before making your changes, but could this be better handled by having the API controllers inherit from a different controller that does not enforce basic auth?  

Example:  https://github.com/Tahi-project/ihat/blob/master/app/controllers/admin/jobs_controller.rb#L2
And: https://github.com/Tahi-project/ihat/blob/master/app/controllers/admin_controller.rb